### PR TITLE
VB-3233 Add configuration structure to DB and Entities server side

### PR DESCRIPTION
**Pull request by AledWEvans**
Add configuration structure to DB and Entities server side

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonConfigService.kt [hunk 1]
```diff
@@ -115,4 +134,32 @@ class PrisonConfigService(
...
     return prison.excludeDates.stream().map { it.excludeDate }.collect(Collectors.toSet())
   }
+
+  private fun validatePrisonDetails(
+    updatePolicyNoticeDaysMin: Int,
+    policyNoticeDaysMin: Int,
+    policyNoticeDaysMax: Int,
+    prisonCode: String,
+  ) {
+    if (updatePolicyNoticeDaysMin > policyNoticeDaysMin) {
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonConfigService.kt:144]: Need to write a test for these validations, also min value 0 might be illegal on normal min but not update

Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonConfigService.kt:144]: added this validation now

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/admin/PrisonAdminController.kt [hunk 2]
```diff
@@ -137,6 +138,46 @@ class PrisonAdminController(
...
+        description = "Prison created",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to create prison",
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/admin/PrisonAdminController.kt:166]: description - update prison

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonConfigService.kt [hunk 3]
```diff
@@ -115,4 +132,21 @@ class PrisonConfigService(
   private fun getExistingExcludeDates(prison: Prison): Set<LocalDate> {
     return prison.excludeDates.stream().map { it.excludeDate }.collect(Collectors.toSet())
   }
+
+  private fun validatePrisonDetails(
+    policyNoticeDaysMin: Int,
+    policyNoticeDaysMax: Int,
+    prisonCode: String,
+  ) {
+    if (policyNoticeDaysMin > policyNoticeDaysMax) {
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonConfigService.kt:141]: add validation for > 0?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/UpdatePrisonDto.kt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.dto
+
+import io.swagger.v3.oas.annotations.media.Schema
+
+@Schema(description = "Prison update dto")
+data class UpdatePrisonDto(
+  @Schema(description = "minimum number of days notice from the current date to booked a visit", example = "2", required = false)
+  var policyNoticeDaysMin: Int?,
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/UpdatePrisonDto.kt:8]: should both values be mandatory during an update?

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**